src/blackmetal/src/communication/BlackMetal.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "Client.hpp"
+
+#include <string>
+#include <memory>
+
+#include <rclcpp/rclcpp.hpp>
+#include "geometry_msgs/msg/twist.hpp"
+
+/**
+ * @class BlackMetal
+ * @brief Singleton class for communication with the blackmetal robot.
+ *
+ * The class creates client and returns this instance on every call.
+ */
+class BlackMetal
+	: public rclcpp::Node
+	, public Client
```

Comment:
nebolo by vhodnejsie tieto objekty mat ako private members?

Reply:
Ci to je povinnost pri ros2?

Reply:
Co som pozeral viacero probramov tak vsade sa inheritencia riesila cez public. Pozrel som si aj tak Nodu ze co by to ovplyvnilo a je tam jeden konstruktor, ktory by sa stal privatnym.